muikku-core-plugins/src/main/resources/META-INF/resources/scripts/gui/communicator.js

```diff
@@ -472,6 +476,89 @@ $(document).ready(function(){
     
     
   },
+  
+  _onMoreClick : function(event){
+    var element = $(event.target);
+    var box = window.location.hash.substring(1);;
+    element = element.parents(".cm-messages-paging");
+    var pageElement = $(".cm-messages-pages");
+    var _this = this;  
+
+    $(element).remove(); 
+    _this._addLoading(pageElement);
+    
+
+    var msgsCount = 0;
+    var msgs = $(CommunicatorImpl.msgContainer).find('.cm-message');
+    
+    for(var m = 0; m < msgs.length; m++){
+      msgsCount ++;
+     }
+          
+    var fRes = msgsCount;
+
+
+    switch(box) {
+      case "sent":
+        mApi().communicator.sentitems.read({'firstResult' : fRes})
+        .on('$', function (item, itemCallback) {
+          item.caption = $('<div>').html(item.caption).text();
+          item.content = $('<div>').html(item.content).text();
+          
+          // Lets fetch message recipients by their ids
+          var recipients = item.recipientIds;
+          var recipientNames = [];
+          for (var i = 0; i < recipients.length; i++) {
```

Comment:
This code will work only with extreme luck, first of all theres a mApi call inside a for -loop, which makes callbacks be called asynchronously adding recipients to list in random order, secondly itemCallback is called outside mApi callbacks, so this callback will quite probably will be executed after result has already been processed.